### PR TITLE
Cruise control should not stuck when broker dies during partition movement.

### PR DESCRIPTION
**Pull request by becketqin**

### CODE REVIEW

File: cruise-control/src/main/scala/com/linkedin/kafka/cruisecontrol/executor/ExecutorUtils.scala [hunk 1]
```diff
@@ -86,6 +116,7 @@ object ExecutorUtils {
 
       // We do not use the ReassignPartitionsCommand here because we want to have incremental partition movement.
       if (newPartitionAssignment.nonEmpty)
+        System.out.println(newPartitionAssignment)
```
Comment (efeg) [inline cruise-control/src/main/scala/com/linkedin/kafka/cruisecontrol/executor/ExecutorUtils.scala]: Why do we need to print this? -- i.e. why don't we log it?

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorTest.java [hunk 2]
```diff
@@ -115,6 +115,39 @@ public void testMoveNonExistingPartition() {
...
+      }
+    }
+    int initialLeader0 = (Integer) zkUtils.getLeaderForPartition(topic0, partition).get();
+    int initialLeader1 = (Integer) zkUtils.getLeaderForPartition(topic1, partition).get();
+    _brokers.get(initialLeader0 == 0 ? 1 : 0).shutdown();
+    BalancingProposal proposal0 =
+        new BalancingProposal(tp0, initialLeader0, initialLeader0 == 0 ? 1 : 0, BalancingAction.REPLICA_MOVEMENT);
+    BalancingProposal proposal1 =
+        new BalancingProposal(tp1, initialLeader1, initialLeader1 == 0 ? 1 : 0, BalancingAction.LEADERSHIP_MOVEMENT);
+
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorTest.java]: Should we also tests for REPLICA_DELETION and REPLICA_ADDITION?

Comment (becketqin) [reply cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorTest.java]: Currently those two actions are not generated anywhere. I'll add test when we start to use them.

**Comment by efeg** [comment]
LGTM.

**Pull request merged**